csharp/extractor/Semmle.Extraction.CIL/Entities/DefinitionMethod.cs

```diff
@@ -114,25 +114,39 @@ public override IEnumerable<IExtractionProduct> Contents
 
                     if (!body.LocalSignature.IsNil)
                     {
-                        var locals = Context.MdReader.GetStandaloneSignature(body.LocalSignature);
-                        var localVariableTypes = locals.DecodeLocalSignature(Context.TypeSignatureDecoder, this);
-
-                        this.locals = new LocalVariable[localVariableTypes.Length];
+                        var localVariableTypes = System.Collections.Immutable.ImmutableArray<Type>.Empty;
+                        var hasError = false;
+                        try
+                        {
+                            var locals = Context.MdReader.GetStandaloneSignature(body.LocalSignature);
+                            localVariableTypes = locals.DecodeLocalSignature(Context.TypeSignatureDecoder, this);
+                        }
+                        catch (System.BadImageFormatException exc)
+                        {
+                            Context.Extractor.Logger.Log(Util.Logging.Severity.Info,
+                                $"Could not decode locals in method {declaringType.GetQualifiedName()}.{name}. {exc}");
+                            hasError = true;
+                        }
 
-                        for (var l = 0; l < this.locals.Length; ++l)
+                        if (!hasError)
```

Comment:
I have a slight preference for `if (hasError) return`, as it reduces indentation.

Reply:
We're still able to CIL extract the body even through we had trouble with the local variables, so we can't return early. Also, this method has `yield returns` , which can't be inside try-catch blocks. That's why this `hasError` was introduced. It's ugly, but I didn't have any better idea.

Reply:
Ah, I mistakenly read the `}` on line 152 as the end of the method.